x/rewards/keeper/providers.go

```diff
@@ -121,3 +124,110 @@ func (k Keeper) specProvidersBasePay(ctx sdk.Context, chainID string) ([]types.B
 	}
 	return basepays, totalBasePay
 }
+
+// ContributeToValidatorsAndCommunityPool transfers some of the providers' rewards to the validators and community pool
+// the function return the updated reward after the participation deduction
+func (k Keeper) ContributeToValidatorsAndCommunityPool(ctx sdk.Context, reward math.Int, senderModule string) (updatedReward math.Int, err error) {
+	// calculate validators and community participation fractions
+	validatorsParticipation, communityParticipation, err := k.CalculateContributionPercentages(ctx, reward, senderModule)
+	if err != nil {
+		return reward, err
+	}
+
+	if communityParticipation.Equal(sdk.OneDec()) {
```

Comment:
can it be one?
validatorsParticipation + communityParticipation <= 1 isnt it?

Reply:
In `CalculateContributionPercentages` I check for communityTax == 1, if so, the community participation is 1 and validators participation is zero, so this can happen